### PR TITLE
Fix default quota display value in user row

### LINKED ISSUE #21085: "Default quota" incorrectly displayed in administrator settings
### Steps to reproduce

1. Log in as an administrator.
2. Select from the "Users" settings.
3. Edit user settings.
4. Change "Quota" to "Default quota".
5. Save settings.

### Expected behaviour
Correct display of the "Default quota".

### Actual behaviour
The default limit is displayed as "0 B".

obraz

**Nextcloud version:** (see Nextcloud admin page)
19.0.0 RC2 and 18.0.4

### PR DESCRIPTION
**Pull request by PVince81**
When selecting default quota in the dropdown, the actual numeric value
or "Unlimited" will be displayed instead of "0 B". This matches whatever
is displayed after refreshing the page.

In the case where the default quota is unlimited, for which the server
returns -3 instead of "none", the frontend now also shows "Unlimited"
instead of "0 B".

Fixes https://github.com/nextcloud/server/issues/21085

### CODE REVIEW

File: apps/settings/src/components/UserList/UserRowSimple.vue [hunk 1]
```diff
@@ -154,11 +154,24 @@ export default {
 			return getCurrentUser().uid !== this.user.id || this.settings.isAdmin
 		},
 		userQuota() {
-			if (this.user.quota.quota === 'none') {
-				return t('settings', 'Unlimited')
+			let quota = this.user.quota.quota
+
+			// when the default quota is unlimited, the server returns -3 here, map it to "none"
+			if (quota === -3) {
+				quota = 'none'
```
Comment (skjnldsv) [inline apps/settings/src/components/UserList/UserRowSimple.vue:161]:
```suggestion
			if (quota === -3 || quota === 'none') {
				return t('settings', 'Unlimited')
```

Merge the two?

**Review by skjnldsv — APPROVED** [review]
Nice! :rocket: 
A small comment, good otherwise :wink:

**Comment by PVince81** [comment]
ah yes, my master was not up to date... sorry little bot... fixing

**Comment by PVince81** [comment]
@skjnldsv I've simplified even further, mind having a second look ?

**Review by GretaD — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]
Perfect! :+1:

**Comment by PVince81** [comment]
> The backport to stable20 failed. Please do this backport manually.

on it... all you need to do is recompile everything

**Comment by PVince81** [comment]
stable20 backport here: https://github.com/nextcloud/server/pull/23726

took me a while as I'm often confused at the weird juggling that is happening in the "js/*" folders...

**Comment by MorrisJobke** [comment]
@PVince81 Keep in mind to remove the backport-request label after the backport was opened. In this way we can go over the PRs that still have this label to find pending backports.

**Pull request merged**